core/templates/dev/head/components/HintAndSolutionButtonsDirective.js

```diff
@@ -0,0 +1,95 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for hint and solution buttons.
+ */
+
+oppia.directive('hintAndSolutionButtons', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/components/hint_and_solution_buttons_directive.html'),
+      controller: [
+        '$scope', '$rootScope', 'HintManagerService',
+        'ExplorationPlayerService', 'SolutionManagerService',
+        'PlayerTranscriptService', 'HintAndSolutionModalService',
+        'DeviceInfoService', 'WindowDimensionsService',
+        function(
+          $scope, $rootScope, HintManagerService,
+          ExplorationPlayerService, SolutionManagerService,
+          PlayerTranscriptService, HintAndSolutionModalService,
+          DeviceInfoService, WindowDimensionsService) {
+          $scope.defaultHintUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/default_hint.svg');
+
+          $scope.activeHintUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/active_hint.svg');
+
+          $scope.defaultSolutionUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/default_solution.svg');
+
+          $scope.activeSolutionUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/active_hint.svg');
+
+          $scope.hintPositions = [];
+          $scope.solutionExists = null;
+          $scope.isWindowNarrow = WindowDimensionsService.isWindowNarrow();
+          $scope.activeModalIndex = null;
```

Comment:
Same thing, I don't know what this means.

EDIT: maybe $scope.activeHintIndex? Seems like it's an index into the hints rather than into a bunch of modals. Also we can decide to talk about the currently-viewed hint as the "active hint", that seems natural. Still probably worth a comment, though.